src/main.rs

```diff
@@ -131,12 +173,19 @@ impl State {
     /// `HanoiError::UnstableStack` if this operation attempted to put `disk` on a smaller
     /// disk.
     fn push_disk(&mut self, peg: Peg, disk: Disk) -> Result<(), HanoiError> {
-        unimplemented!()
+        let mut tow = self.get_tower_mut(peg);
+        if (tow.last().is_none() || (disk < *tow.last().unwrap())) {
+            tow.push(disk)
+        }
+
+        Err(HanoiError::UnstableStack(peg, disk))
     }
```

Comment:
Hmm, this function seems to always return an `Err`. Does pushing a disk always cause an error?